### ISSUE TITLE
Package Request: llpp - saner pdf viewer

### ISSUE THREAD

**Issue by etosan** [void-linux/void-packages#29676]
**package name:** llpp

**project homepage:** https://repo.or.cz/w/llpp.git

**purpose:** smooth (scrolling) continuous (pages wrap seamlessly) pdf viewer, with advanced features built on top of mudpf

There also seem to be copy of the repo here on github https://github.com/moosotc/llpp, but repo.cz project is referenced in other package managers. 

This is thick layer built around mupdf core, which makes it better pdf/cbz/etc viewer. Most important features are: it provides continuous "page stream" (you see previous page bottom and next page top when scrolling, in addittion to current one), in addition to that, it has smooth scrolling (non-jumping panning). Finally it supports arbitrary numbers of columns (for example 2 col mode: L page, R page; but even 3+ column modes are possible). All these capabilities make it easier on the eyes when reading "serious" books on the computer.

Unfortunately the outer non mupdf part is written in OCaml lang. - I have no clue how ocaml works and how this can be packaged into the void. Maybe some of our OCaml experts could chime in?

**Comment by mattheusroxas** [comment]
That looks pretty cool

**Comment by ericonr** [comment]
```

outd="$1"
srcd="$(dirname $0)"
mudir=$outd/mupdf
muinc="-I $mudir/include -I $mudir/thirdparty/freetype/include"

test -d "$mudir" || die muPDF not found, consult $(dirname $0)/BUILDING
```

This doesn't look like the sort of package that actually wants to use system deps...

Arch even adds their own Makefile on top lol  https://github.com/archlinux/svntogit-community/blob/52005944de09401d344cfd4ee62581b42229bb03/trunk/GNUmakefile

I'm not keen on packaging this.

**Comment by ericonr** [comment]
You might want to try zathura as a PDF viewer...

**Comment by eli-schwartz** [comment]
My Makefile for Arch uses system deps just fine. :D It should be pretty simple to compile if you use said Makefile.

Gentoo uses it now too, these days. Before that, they repeatedly ran ocamlc in their ebuild. :/

**Comment by etosan** [comment]
@ericonr 
> You might want to try zathura as a PDF viewer...

I remember trying `zathura` around 5+ years ago and it was very slow, quite heavy and quite cumbersome (nothing against it I am just not a fan). I am okay with few vimmy commands, but I am not really a friend with heavy vimmyness (if I remember correctly, `zathura` was very vimmy).

`mupdf` on the other hand is very lean, fast and light (I would say it is `mplayer` of pdf viewers). But doing extended reading sessions in it is getting pretty hard on me, as as I am getting older the pageflips are desorienting me too much.

Fortunately, then I found `llpp` and it was love at first sight. It has exactly the features I need and I care about (a lot), that no other pdf viewer provides (if I am not mistaken), so if `mupdf` is `mplayer`, `llpp` is `mpv` of pdf viewers. I don't see zathura as useful in my setup.

@eli-schwartz: yes I have my own build installed to /usr/local, but I requested this here, as it would allow me to get rid of the minimal maintenance it needs. It would perhaps help others as well as it did for me.

I have no clue how OCaml packaging, compiling and system interaction with libraries works, so that is why I have no idea, what it would entail, to have it as proper package.

Is that such a problem that mupdf is amalgammated into it statically?

**Comment by calibius** [comment]
The problem is the current mupdf package doesn't provide a mupdf.pc and there's no (ocaml-)lablgl package. So you're looking at multiple packages/changes to support this one if you want to use the makefile. I've looked at it, but I'm not knowledgeable enough to do it.  You can check out my fork and maybe point me in the right direction? It's kind of messy at the moment lol ...

**Comment by etosan** [comment]
I am by no means expert but do I understand it right that .pc is pkgconf (or pkg-config?) extension that is used to provide compile and link time options to -dev packages? I though all of void packages provide that file, but mupdf does not?

**Comment by calibius** [comment]
correct, mupdf doesn't include a .pc file in the package because the developer doesn't; other distros add it back at build time. Most packages provide it because they're provided by the developer, is my understanding.

**Comment by etosan** [comment]
So with mupdf we would have to provide it ourselves?
